dashboard/app/models/course_offering.rb

```diff
@@ -19,20 +19,22 @@
 #  school_subject                   :string(255)
 #  device_compatibility             :string(255)
 #  description                      :string(255)
-#  self_paced_professional_learning :string(255)
 #  professional_learning_program    :string(255)
 #  video                            :string(255)
 #  published_date                   :datetime
+#  self_paced_pl_course_offering_id :integer
 #
 # Indexes
 #
+#  fk_rails_8283d50b87            (self_paced_pl_course_offering_id)
 #  index_course_offerings_on_key  (key) UNIQUE
 #
 
 class CourseOffering < ApplicationRecord
   include Curriculum::SharedCourseConstants
 
   has_many :course_versions, -> {where(content_root_type: ['UnitGroup', 'Unit'])}
+  has_one :self_paced_pl_course_offering, class_name: 'CourseOffering', foreign_key: :self_paced_pl_course_offering_id
```

Comment:
I just had to read this approximately 17 times, but I think this should be a `belongs_to` association: https://guides.rubyonrails.org/association_basics.html#the-belongs-to-association

Reply:
Ah wait, I'm realizing we're having a model refer to something in the same model ... I have no idea how this works. I'll wait for others to help!

Reply:
I agree with Meg that this should be `belongs_to`.

self-joins are legit and and documented here: https://guides.rubyonrails.org/association_basics.html#self-joins

since the reference name matches the db field name (minus the `_id` part), it might be possible to shorten this to:
```suggestion
  belongs_to :self_paced_pl_course_offering, class_name: 'CourseOffering'
```

However  @juanmanzojr  I think it would be best if you could try this out in the rails console to confirm that it does what you expect before merging.

Reply:
Ill give this a try! Thank you both!

Reply:
Thank you, Dave, for the link to learn about self-joins!